pkg/codegen/go/gen.go

```diff
@@ -4687,8 +4687,11 @@ func GeneratePackage(tool string,
 
 	files := codegen.Fs{}
 
-	// If the package is parameterized generate a go.mod for it
-	if pkg.Parameterization != nil {
+	// If the package is parameterized generate a go.mod for it iff it is not a
+	// part of another package (as in the case for a paramaterized provider).
+	// If there are localDependencies, this is an indicates a go.mod is still
+	// required and this is part of a package add.
+	if pkg.Parameterization != nil && len(localDependencies) == 0 {
```

Comment:
I'm not sure I follow this? We eventually want to be generating go.mod files for all packages. Why would we only generate the go mod if localDeps is empty? Doesn't that mean we skip the go.mod for the conformance test (because localDeps will have the pulumi local dep in it)? But it's still committed in this PR?

Reply:
So what's happening in go sdk-gen is that we are generating _two_ go.mod files, one for the top-level package which has correct references to local sdks with replacements (the one committed in this PR) and another go.mod file underneath the root go _module_ which is what is being skipped in this PR. See files in previous attempt https://github.com/pulumi/pulumi/pull/17795 where there are two go.mod files which I believed what caused `go mod tidy` to fail for this conformance test. 

This second go.mod file we are skipping here seems only relevant for generating go.mod during `pulumi package add ...`

Reply:
I like this as well, changed it